tests/e2e/ctl_v3_auth_cluster_test.go

```diff
@@ -0,0 +1,113 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/tests/v3/framework/config"
+	"go.etcd.io/etcd/tests/v3/framework/e2e"
+)
+
+func TestAuthCluster(t *testing.T) {
+	e2e.BeforeTest(t)
+	cfg := &e2e.EtcdProcessClusterConfig{
+		ClusterSize:   1,
+		InitialToken:  "new",
+		SnapshotCount: 2,
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	epc, err := e2e.NewEtcdProcessCluster(ctx, t, cfg)
+	if err != nil {
+		t.Fatalf("could not start etcd process cluster (%v)", err)
+	}
+	defer func() {
+		if err := epc.Close(); err != nil {
+			t.Fatalf("could not close test cluster (%v)", err)
+		}
+	}()
+
+	epcClient := epc.Client()
+	createUsers(ctx, t, epcClient)
+
+	if err := epcClient.AuthEnable(ctx); err != nil {
+		t.Fatalf("could not enable Auth: (%v)", err)
+	}
+
+	numKeys := 10
+	testUserClientOpts := e2e.WithAuth("test", "testPassword")
+	rootUserClientOpts := e2e.WithAuth("root", "rootPassword")
+
+	// write more than SnapshotCount keys to single leader to make sure snapshot is created
+	for i := 0; i <= numKeys; i++ {
+		if err := epc.Client(testUserClientOpts).Put(ctx, fmt.Sprintf("/test/%d", i), "test", config.PutOptions{}); err != nil {
+			t.Fatalf("failed to Put (%v)", err)
+		}
+	}
+
+	// start second process
+	if err := epc.StartNewProc(ctx, t, rootUserClientOpts); err != nil {
+		t.Fatalf("could not start second etcd process (%v)", err)
+	}
+
+	// make sure writes to both endpoints are successful
+	for _, endpoint := range epc.EndpointsV3() {
+		if err := epc.Client(testUserClientOpts, e2e.WithEndpoints([]string{endpoint})).Put(ctx, "/test/key", endpoint, config.PutOptions{}); err != nil {
+			t.Fatalf("failed to write to Put to %q (%v)", endpoint, err)
+		}
+		numKeys++
+	}
+
+	// verify cluster state is consistent
+	hashKvs, err := epc.Client(rootUserClientOpts).HashKV(ctx, int64(numKeys))
```

Comment:
The latest revision might not be `numKeys`.

```suggestion
	hashKvs, err := epc.Client(rootUserClientOpts).HashKV(ctx, 0)
```